model/src/main/java/org/perfrepo/dto/SearchResult.java

```diff
@@ -0,0 +1,45 @@
+package org.perfrepo.dto;
+
+import java.util.List;
+
+/**
+ * Wrapper object for search query with pagination support.
+ *
+ * @author Jiri Grunwald (grunwjir@gmail.com)
+ */
+public class SearchResult<T> {
+    private List<T> data;
+    private Integer totalCount;
```

Comment:
I think in all 4 coming cases, it doesn't make sense to have it Integer, we should use int instead. It has a benefit of eliminating the pesky NPE possibility. Or you have any reason why not to do that? Are there any cases where the null value would signalize some use case?